lib/src/extended_file_image_provider.dart

```diff
@@ -1,56 +1,9 @@
-import 'dart:io';
-import 'dart:typed_data';
-import 'dart:ui' as ui show Codec;
-import 'package:flutter/widgets.dart';
-
-import 'extended_image_provider.dart';
-
-class ExtendedFileImageProvider extends FileImage with ExtendedImageProvider {
-  ExtendedFileImageProvider(
-    File file, {
-    double scale = 1.0,
-  }) : super(file, scale: scale);
-
-  @override
-  ImageStreamCompleter load(FileImage key, DecoderCallback decode) {
-    return MultiFrameImageStreamCompleter(
-      codec: _loadAsync(key, decode),
-      scale: key.scale,
-      debugLabel: key.file.path,
-      informationCollector: () sync* {
-        yield ErrorDescription('Path: ${file.path}');
-      },
-    );
-  }
-
-  Future<ui.Codec> _loadAsync(FileImage key, DecoderCallback decode) async {
-    assert(key == this);
-
-    final Uint8List bytes = await file.readAsBytes();
-
-    if (bytes.lengthInBytes == 0) {
-      // The file may become available later.
-      PaintingBinding.instance!.imageCache!.evict(key);
-      throw StateError('$file is empty and cannot be loaded as an image.');
-    }
-
-    return await instantiateImageCodec(bytes, decode);
-  }
-
-  @override
-  bool operator ==(dynamic other) {
-    if (other.runtimeType != runtimeType) {
-      return false;
-    }
-    if (other is ExtendedFileImageProvider &&
-        file.path == other.file.path &&
-        scale == other.scale) {
-      imageData.data ??= other.rawImageData;
-      return true;
-    }
-    return false;
-  }
-
-  @override
-  int get hashCode => hashValues(file.path, scale);
+import 'io/extended_file_image_provider.dart'
+    if (dart.library.html) 'extended_file_image_provider.dart'
+    as image_provider;
+
+class ExtendedFileImageProvider
+    extends image_provider.ExtendedFileImageProvider {
+  ExtendedFileImageProvider(Object file, {double scale = 1.0})
```

Comment:
Discussed offline, we don't think `Object` is good enough here, since it's a file image provider.

Reply:
Maybe, but `File` imports  from dart:io & pub analyzer for web cannot support import from dart:io.
I don't see any other possible solution.

But to improve support, can add a assert for a io file type with expanded information here:
https://github.com/SergeShkurko/extended_image_library/blob/41c1dd1dfb7ae789ee30261ad77576047ade6f89/lib/src/io/extended_file_image_provider.dart#L10
If developer pass other type he see error at console

the advantage of this method is theoretical support html.File at web

Reply:
Took a look about the `path_provider`'s score on pub.dev, it seems not support all platforms at present, I'm wondering the change won't take any effect once we import it. https://pub.dev/packages/path_provider/score

Reply:
the workaround still works. specially published out for verification: https://github.com/fluttercandies/extended_image_library/pull/26#issuecomment-768220707

https://pub.dev/packages/extended_image_library_web_test/score

Reply:
IMO it's more like a regression than a workaround, because we lost the static type safety when using `Object`.

Just curious, what point will you give from 0 to 10 if you evaluate the effect of your package by this "unsupported platform"?